### PR TITLE
Fixing Readme

### PR DESCRIPTION
**Pull request by mrm9084**
## Description
Fixing outdated App Config Readme.

### CODE REVIEW

File: spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md [hunk 1]
```diff
@@ -193,19 +193,19 @@ public class MyCredentials implements AppConfigCredentialProvider, KeyVaultCrede
 
 ### Modifying Connection Client
```
Comment (zhenlan) [inline spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:194]: How about "Client Builder Customization"?

File: spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md [hunk 2]
```diff
@@ -193,19 +193,19 @@ public class MyCredentials implements AppConfigCredentialProvider, KeyVaultCrede
...
 
-In some situations the client connection needs to be modified, such as adding proxy configurations. Using the AppConfigurationClientProvider and/or KeyVaultClientProvider. By implementing either of these classes and providing and generating a @Bean of them will enable client modification following [App Configuration SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/appconfiguration/azure-data-appconfiguration#key-concepts) and [Key Vault SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-secrets#key-concepts).
+In some situations the client connection needs to be modified, such as adding proxy configurations. Using the ConfigurationClientBuilderSetup and/or SecretClientBuilderSetup. By implementing either of these classes and providing and generating a @Bean of them will enable client modification following [App Configuration SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/appconfiguration/azure-data-appconfiguration#key-concepts) and [Key Vault SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-secrets#key-concepts).
```
Comment (zhenlan) [inline spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:196]: How about this?

The service client builders used for connecting to App Configuration and Key Vault can be customized by implementing interfaces `ConfigurationClientBuilderSetup` and `SecretClientBuilderSetup` respectively. Generating and providing a `@Bean` of them will update the default service client builders used in [App Configuration SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/appconfiguration/azure-data-appconfiguration#key-concepts) and [Key Vault SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-secrets#key-concepts). If necessary, the customization can be done per App Configuration store or Key Vault instance.
```java
public interface ConfigurationClientBuilderSetup {
    public void setup(ConfigurationClientBuilder builder, String endpoint);
}

public interface SecretClientBuilderSetup {
    public void setup(SecretClientBuilder builder, String uri);
}
```
For example, the following implementation of `MyClient` replaces the default `HttpClient` with one using a proxy for all traffic to App Configuration and Key Vault.
```java
public class MyClients implements ConfigurationClientBuilderSetup, SecretClientBuilderSetup {
    ... ...
}
```

File: spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md [hunk 3]
```diff
@@ -193,19 +193,19 @@ public class MyCredentials implements AppConfigCredentialProvider, KeyVaultCrede
...
     }
 
     @Override
-    public ConfigurationClientBuilder modifyConfigurationClient(ConfigurationClientBuilder configurationClientBuilder) {
-        return configurationClientBuilder.httpClient(buildHttpClient());
+    public void setup(ConfigurationClientBuilder builder, String endpoint) {
+        builder.httpClient(buildHttpClient());
     }
 
     private HttpClient buildHttpClient() {
```
Comment (zhenlan) [inline spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:211]: Can the SDK pick up settings `https.proxyHosts` and `https.proxyPort` automatically?

Comment (mrm9084) [reply spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:211]: It does, but only if we created our own HttpClient like we do in the sample. The issue is that we would either have to choose Netty or Okhttp, which could cause issues with the forced dependency.

File: spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md [hunk 4]
```diff
@@ -191,21 +191,33 @@ public class MyCredentials implements AppConfigCredentialProvider, KeyVaultCrede
...
-    public SecretClientBuilder modifyKeyVaultClient(SecretClientBuilder keyVaultClientBuilder) {
-        return keyVaultClientBuilder.httpClient(buildHttpClient());
+    public void setup(SecretClientBuilder builder, String uri) {
+        builder.httpClient(buildHttpClient());
     }
 
     @Override
-    public ConfigurationClientBuilder modifyConfigurationClient(ConfigurationClientBuilder configurationClientBuilder) {
-        return configurationClientBuilder.httpClient(buildHttpClient());
+    public void setup(ConfigurationClientBuilder builder, String endpoint) {
```
Comment (zhenlan) [inline spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:219]: Can we switch the order of this and the one for key vault so we follow the same order of interfaces above?

File: spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md (hunk 3 above)
Comment (zhenlan) [reply spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md:223]: In the default implementation, isn't the SDK already made the choice of Netty vs. Okhttp?

**Pull request merged**